File: javascript/raindrops/raindrops.js
// raindrops.js

export const convert = number => {
  var sound = "";

  if (isDivisible(number, 3)) {
    sound += "Pling";
  }

  if (isDivisible(number, 5)) {
    sound += "Plang";
  }

  if (isDivisible(number, 7)) {
    sound += "Plong";
  }

  if (sound == "") {
    sound = number.toString();
  }

  return sound;
};

const isDivisible = (number, divisor) => number % divisor == 0;

File: cpp/raindrops/raindrops.cpp
#include "raindrops.h"


// function definitions ------------------------------------------------------

namespace raindrops {
    std::string convert(int n)
    {
        std::string sound = "";

        if (n % 3 == 0) {
            sound.append("Pling");
        }

        if (n % 5 == 0) {
            sound.append("Plang");
        }

        if (n % 7 == 0) {
            sound.append("Plong");
        }

        if (sound.empty()) {
            sound = std::to_string(n);
        }

        return sound;
    }
}  // namespace raindrops
